src/components/Geo/Geo.tsx

```diff
@@ -1,14 +1,93 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { Helmet } from 'react-helmet'
+import { SpeciesSelect } from './SpeciesSelect'
 import { MapPlot } from './MapPlot'
+import { TimePlot } from './TimePlot'
 import { SelectionInfo } from './SelectionInfo'
 import { RunData } from './types'
 import { helpIcon } from 'common'
+import * as d3 from 'd3'
+import rdrpPosTsv from './rdrp_pos.tsv'
+import { DateTime } from 'luxon'
+
+async function retrieveAllPossibleRows(
+    inputFile: any,
+    setAllPossibleRows: React.Dispatch<React.SetStateAction<RunData[] | undefined>>
+) {
+    let rows = ((await d3.tsv(inputFile)) as object) as RunData[]
+    setAllPossibleRows(rows)
+}
+
+/**
+ * This component:
+ * - Begins by reading in all the possible rows contained in the rdrpPosTsv file
+ * - Gets all the unique species to pass to the multi-select dropdown
+ * - Creates an object of release dates (months) to number of species for all rows
+ * - Returns the components of the Geo page
+ */
 
 export const Geo = () => {
+    const [allPossibleRows, setAllPossibleRows] = React.useState<RunData[]>()
+    const [allUniqueSpecies, setAllUniqueSpecies] = React.useState<string[]>()
+    const [allRowsTimePlot, setAllRowsTimePlot] = React.useState<{ [key: string]: number }>()
+
     const [selectedPoints, setSelectedPoints] = React.useState<RunData[]>()
+    const [selectedSpecies, setSelectedSpecies] = React.useState<String[]>()
+    const [selectedRows, setSelectedRows] = React.useState<RunData[]>()
     const [isCollapsed, setIsCollapsed] = React.useState<boolean>(false)
 
+    // Retrieve all possible rows from the rdrpPosTsv file
+    if (allPossibleRows === undefined) retrieveAllPossibleRows(rdrpPosTsv, setAllPossibleRows)
+    // From all of the possible rows, distill the unique species to be passed into the SpeciesSelect component
```

Comment:
What do you think about pre-computing this "Sort Unique" operation and serving a second file of the species list alongside `rdrpPosTsv`?

Ideally both `rdrpPosTsv` and the unique species list should be an API call to the server, but this hasn't been built yet.

(I'm not saying to do this, just what your thoughts are)

Reply:
I'll chime in here – this is definitely a good direction, and requires modifications across the whole stack. Writing some docs on how to manage the database and query it via the API is on my todo-list – it should be helpful here! I wouldn't worry yet unless the sort is causing significant performance issues.

Reply:
Hi, I also believe this is in the right direction.  I chose not to place the unique species into a separate file because:

1. It seemed like querying an API instead of the rdrpPosTsv file would be the eventual goal, and
2. If the rdrpPosTsv file gets updated/added to every so often, a process would have to be put in place to also update the unique species file.

Certainly a conversation to keep in mind going forwards.

Reply:
I like that thinking :) :+1:  For sure API is the way  to do this "properly" this was all a bit of a hack when we were putting it together.